### PR TITLE
update openstack sdk

### PR DESCRIPTION
**Pull request by icewind1991**
- [x] requires https://github.com/nextcloud/server/pull/8259
- [x] requires https://github.com/nextcloud/3rdparty/pull/85

Required for enabling v3 authentication

### CODE REVIEW

**Comment by MorrisJobke** [comment]
Conflicts :/

**Comment by icewind1991** [comment]
All rebased

**Review by MorrisJobke — APPROVED** [review]

**Comment by rullzer** [comment]
I rebased as I want the swift tests to pass here :P

**Comment by icewind1991** [comment]
The swift docker image is being a PITA, I'm trying to get it more reliable

**Review by rullzer — APPROVED** [review]

**Comment by MorrisJobke** [comment]
@oparoz No backport of this - it's just too much for a backport.

@rullzer Could you backport your last commit to stable13 to have better CI experience on stable13 as well?

**Comment by rullzer** [comment]
@MorrisJobke we don't run swift tests on stable13

File: apps/files_external/lib/Lib/Storage/Swift.php [hunk 1]
```diff
@@ -553,6 +537,7 @@ public function rename($path1, $path2) {
 
 			// cleanup
 			if ($this->unlink($path1) === false) {
+				throw new \Exception('failed to remove original');
```
Comment (MorrisJobke) [inline apps/files_external/lib/Lib/Storage/Swift.php]: The two lines after this are unreachable 🙈

**Pull request merged**